### PR TITLE
feat(): changing user password

### PR DESCRIPTION
**Pull request by Izak88**

### CODE REVIEW

File: src/app/services/auth.service.ts [hunk 1]
```diff
@@ -63,6 +63,34 @@ export class AuthService {
       .toPromise()
       .then(data => data);
   }
+
+  changePassword(oldpass: string, pass1: string, pass2: string): Promise<any> {
+    return new Promise((resolve, reject) => {
+      if (pass1 !== pass2) {
+        reject(`New password doesn't match!`);
```
Comment (jkuri) [inline src/app/services/auth.service.ts:82]: I would make that as a validator, for now we can merge this and I can update later then, okay?

File: src/app/components/app-settings/app-settings.component.ts [hunk 2]
```diff
@@ -1,20 +1,75 @@
...
   selector: 'app-settings',
   templateUrl: 'app-settings.component.html'
 })
 export class AppSettingsComponent implements OnInit {
-  user: any;
+  tab: string;
+  success: boolean;
+  error: boolean;
+  errorMessage: string;
+  oldpassword: string;
```
Comment (jkuri) [inline src/app/components/app-settings/app-settings.component.ts]: can you make `interface` for this? i.e.
```ts
form: { oldPassword: string, newPassword: string, repeatPassword: string };
```
its more convenient.

File: src/app/services/auth.service.ts (hunk 1 above)
Comment (Izak88) [reply src/app/services/auth.service.ts:82]: Can you validate that directly in template or did you mean in a function in component? Yeah, we can update that later...

Comment (jkuri) [reply src/app/services/auth.service.ts:82]: you can create custom validators, see https://angular.io/guide/form-validation

File: src/api/router.ts [hunk 3]
```diff
@@ -28,6 +28,8 @@ router.get('/api/package/get-random', web.getRandomPackages);
 router.get('/api/package/search+*', web.searchPackages);
 router.get('/api/package/:package(*)', web.getPackage);
 router.post('/api/user/login', web.login);
+router.post('/api/user/changepass', web.changePassword);
```
Comment (jkuri) [inline src/api/router.ts]: name routes with kebab-case, i.e. `/api/user/change-password`.

File: src/app/components/app-settings/app-settings.component.ts (hunk 2 above)
Comment (Izak88) [reply src/app/components/app-settings/app-settings.component.ts]: done

File: src/api/router.ts (hunk 3 above)
Comment (Izak88) [reply src/api/router.ts]: done

**Review by jkuri — APPROVED** [review]

**Pull request merged**